tools/stress-cluster/services/Stress.Watcher/src/JobEventHandler.cs

```diff
@@ -0,0 +1,240 @@
+                return false;
+            }
+
+            var initContainers = job.Spec?.Template?.Spec?.InitContainers;
+            if (initContainers == null || initContainers.Count() == 0) {
+                Logger.Debug($"No deploy init container found, skipping resource deletion.");
+                return false;
+            }
+
+            var deployContainers = initContainers.Where(c => c.Name == "init-azure-deployer");
+            if (deployContainers.Count() == 0)
+            {
+                Logger.Debug($"No deploy init container found, skipping resource deletion.");
+                return false;
+            }
+
+            /*
+               Check if job is done by determing whether any conditions are satifisfied as "True"
+               This is most flexible because checking succeeded/failed counts will not be flexible
+               when there are multiple pods specified for the job.
+
+               Example succeeded state
+               --------------------
+                 status:
+                   completionTime: "2022-10-26T20:53:20Z"
+                   conditions:
+                   - lastProbeTime: "2022-10-26T20:53:20Z"
+                     lastTransitionTime: "2022-10-26T20:53:20Z"
+                     status: "True"
+                     type: Complete
+                   ready: 0
+                   startTime: "2022-10-26T20:52:28Z"
+                   succeeded: 1
+               --------------------
+
+               Example failed state
+               --------------------
+                 status:
+                    conditions:
+                    - lastProbeTime: "2022-10-26T21:05:39Z"
+                      lastTransitionTime: "2022-10-26T21:05:39Z"
+                      message: Job has reached the specified backoff limit
+                      reason: BackoffLimitExceeded
+                      status: "True"
+                      type: Failed
+                    failed: 1
+                    ready: 0
+                    startTime: "2022-10-26T21:04:58Z"
+               --------------------
+           */
+            var isCompleted = job.Status?.Conditions?.Any(c => c.Status == "True");
+            if (isCompleted != true)
+            {
+                Logger.Debug($"Job is not completed, skipping resource deletion.");
+                return false;
+            }
+
+            // NOTE: this will not handle labels added to a pod after deployment
+            job.Metadata.Labels.TryGetValue("Skip.RemoveTestResources", out var skipJobRemove);
+            job.Spec.Template.Metadata.Labels.TryGetValue("Skip.RemoveTestResources", out var skipPodRemove);
+
+            if (skipJobRemove == "true" || skipPodRemove == "true")
+            {
+                Logger.Information($"Resource has Skip.RemoveTestResources=true label, skipping resource deletion.");
+                return false;
+            }
+
+            return true;
```

Comment:
so in this case we do want to delete it even if job status is unknown? (isComplete==null)

Reply:
Similar to your earlier comment, I do a strict filter on `isCompleted != true` earlier, so we will only hit this case if it's `true`.

Reply:
oh lol missed that, LGTM!